### ISSUE TITLE
implement return types for methods

### ISSUE THREAD

**Issue by mgrip** [prettier/plugin-php#45]
input:
```php
function returnTypeTest(): string {
  return 'hi';
}

class Test
{
  public function returnTypeTest(): string {
    return 'hi';
  }
}
```
output:
```php
function returnTypeTest() {
  return 'hi';
}
 
class Test
{
  
    public function returnTypeTest()
    {
       return 'hi';
    }
 }
```

**Comment by nicoder** [comment]
This came to mind when seeing the output:

The opening brace of a function should be on its own line according to [PSR-12](https://github.com/php-fig/fig-standards/blob/master/proposed/extended-coding-style-guide.md#44-methods-and-functions) (same as class methods).

(but the opening brace of a [closure](https://github.com/php-fig/fig-standards/blob/master/proposed/extended-coding-style-guide.md#7-closures) goes on the same line as the closing parens of the parameter list)

**Comment by MichaelDeBoey** [comment]
Like @nicoder said, output should be:

```php
function returnTypeTest(): string
{
  return 'hi';
}
 
class Test
{
  
    public function returnTypeTest() : string
    {
       return 'hi';
    }
 }
```

And
```php
$returnTypeTest = function(): string {
  return 'hi';
}
```
Should output
```php
$returnTypeTest = function (): string {
  return 'hi';
}
```

**Comment by mgrip** [comment]
Good catch! Added https://github.com/prettier/plugin-php/issues/54 to deal with the formatting issue - closing this one as the original issue with type hinting was fixed with #53
